File: code/shawn/lab14-pick6/pick6.py
# Shawn Stolsig
# PDX Code Guild 
# Assignment: Lab 14 - Pick 6
# Date: 10/29/2019

import random

# define dictionary for looking up winnings
winnings = dict([(0,0), (1,4), (2,7), (3,100), (4,50000), (5,1000000), (6,25000000)])

# function for generating list of 6 random numbers
def pick6():
    ''' takes no arguments, returns a list of 6 ints each between 1 and 99 '''
    return_list = []
    for i in range(0,6):
        return_list.append(random.randint(1,99))
    return return_list

# function for comparing two lists of numbers
def compare_numbers(num_list1, num_list2):
    ''' function takes two lists of six numbers and returns how many in a row are the same '''
    counter = 0
    while num_list1[counter] == num_list2[counter]:
        counter += 1
    return counter

# Generate a list of 6 random numbers representing the winning tickets
winning_nums = pick6()

# Start your balance at 0
balance = 0
earnings = 0
loops = 100000
expenses = 2 * loops

# Loop 100,000 times, for each loop:
for count in range(0,loops):
    # Generate a list of 6 random numbers representing the ticket
    ticket = pick6()
    # Subtract 2 from your balance (you bought a ticket)
    balance -= 2
    # Find how many numbers match
    matched_nums = compare_numbers(winning_nums,ticket)

    # print a message when you have two or more in a row
    if winnings[matched_nums] > 4:
        print(f'winning ticket: {winning_nums} this ticket: {ticket}  **** winnings of: ${winnings[matched_nums]}')
    # track sum of winnings seperate from balance
    earnings += winnings[matched_nums]
    # Add to your balance the winnings from your matches
    balance += winnings[matched_nums]

# After the loop, print the final balance
print(f'you won: ${earnings} but after buying tickets your balance is now: ${balance}')
print(f'your ROI is: {round(((earnings - expenses) / expenses )*100,2)}%')

File: code/casey/javascript_labs/lab_1b/pick_3/pick_6/pick_6.js
/* Lab 1: Pick 6 - Version 1

Purpose/goal: Have the computer play pick6 many times and determine net balance.

    - Generate a list of 6 random numbers representing the winning ticket - D

    - Start your balance at 0 - D

    - Loop 1000 times, for each loop:
    
        - Generate a list of 6 random numbers representing the ticket - D
    
        - Subtract 2 from your balance (you bought a ticket) - D
    
        - Find how many numbers match - D
    
        - Add to your balance the winnings from your matches - D
    
        - After the loop, print the final balance - D 
    */

// define balance
let balance = 0;

// define winner ticket
const winner = Array.from({length: 6}, () => Math.floor(Math.random() * 100));

// pick6 function
function pick6() { 
    return Array.from({length: 6}, () => Math.floor(Math.random() * 100));
}

// create ticket matching function
function num_matches(a, b) {
    var x = 0;
    for (let i=0; i < b.length; ++i) {
        if (a[i] === b[i]) {
            x += 1;
        }
    }
    return x;
}

// create lottery loop 
let lottery = 0 
while (lottery < 1000) {
    balance -= 2;
    ticket = pick6();
    x = num_matches(ticket, winner);
    const win_arr = [0, 4, 7, 100, 5000, 1000000, 25000000];
    balance += win_arr.indexOf(x);
    lottery += 1;
}

document.write(`After playing the lottery 1000 times, you have a blanace of ${balance} dollars.` + "<br>Very sad.");
